src/config/mod.rs

```diff
@@ -154,18 +152,16 @@ create_config! {
 pub fn load_config(
     file_path: Option<&Path>,
     options: Option<&CliOptions>,
-) -> FmtResult<(Config, Option<PathBuf>)> {
+) -> Result<(Config, Option<PathBuf>), Error> {
```

Comment:
My thought process here is that `load_config` is public API and basically all API should have the same error types. I guess the question is, if we were to return the more specific error, would that be useful for the caller? It's possible that knowing specifically that the path we passed in was bad allows to retry with a different path, but I think realistically a client is more likely to recover by abandoning formatting and telling the user - in that case we should have the detail that the config path was bad in our error, but don't need to supply the io errors.

Reply:
load_config can fail because of other reasons, like file locked, parse error, ...
The more info we give to the API user, the best. My reasoning was that io::error is standard, have all the error kind that we want in this case, can be implicitly be converted to a failure and is used only for the config module which could pretty much be its own crate.

To me, because io::Error can be converted to failure::Error, returning always failure::Error isn't bringing any benefit. 

If we want to have the same error everywhere AND meaningful error, we could opt for the failure error kind pattern? That's more involved but long term that's probably better? 

Up to you where we go from there

Reply:
OK, I think we might want to introduce an error kind and abstract over the io error, but lets return io error for now